.github/workflows/e2e-arm64.yaml

```diff
@@ -20,7 +15,7 @@ jobs:
     steps:
       - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
         with:
-          ref: "${{ inputs.etcdBranch }}"
+          ref: "main"
```

Comment:
Rather than hardcoding `main` I suggest we remove the `with:` completely for `actions/checkout` so we use the default behaviour of only a single commit being fetched for the ref/SHA that triggered the workflow.

This will allow us to safely backport these changes to `release-3.5` branch.